### ISSUE TITLE
Error Boundaries with shouldComponentUpdate method?

### ISSUE THREAD

**Issue by ghost** [yannickcr/eslint-plugin-react#1798]
I'm using this code from here: https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html

```js
import React from "react";

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({ hasError: true });
    // You can also log the error to an error reporting service
    logErrorToMyService(error, info);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }
    return this.props.children;
  }
}
```

and I got this error:

[eslint] Component is not optimized. Please add a shouldComponentUpdate method. (react/require-optimization).

What I should do?

I should add this?

```js
shouldComponentUpdate() {
  return true;
}
```

**Comment by ljharb** [comment]
Personally I don't think that rule should ever be enabled.

Since your component is stateful, you shouldn't actually do anything to "optimize" that component.

**Comment by ljharb** [comment]
I suppose you could make it a PureComponent - but overusing PureComponent doesn’t “optimize”, it often slows things down.

If you do a setState in another lifecycle method, does the optimize rule still warn?

**Comment by johnunclesam** [comment]
The rule is still there...

It disappear only if I put this:


```js
  shouldComponentUpdate() {
    return true;
  }
```

**Comment by ljharb** [comment]
I’d just disable the rule; i don’t know why we included it in the first place.
